app/models/cloud_volume_snapshot.rb

```diff
@@ -10,4 +12,16 @@ class CloudVolumeSnapshot < ApplicationRecord
   has_many   :based_volumes, :class_name => 'CloudVolume'
 
   virtual_total :total_based_volumes, :based_volumes
+
+  def self.class_by_ems(ext_management_system)
+    ext_management_system && ext_management_system.class::CloudVolumeSnapshot
+  end
+
+  def self.my_zone(ems)
+    ems ? ems.my_zone : MiqServer.my_zone
```

Comment:
I am not really sure what it does, so not sure what is the right way, maybe @Fryguy will know?